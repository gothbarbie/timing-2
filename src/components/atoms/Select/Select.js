import React from 'react'
import styled from 'styled-components'

const Option = styled.option``
const SelectStyle = styled.select`
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.colors.gray};
  padding: 1rem 1.5rem;
  font-size: 1.6rem;
  color: ${({ theme }) => theme.colors.dark};
  appearance: none;
  line-height: normal;
  outline: none;
  position: relative;
  background-position: right 1rem top 50%;
  background-repeat: no-repeat;
  background-image: url(data:image/svg+xml;utf8;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pg0KPHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIiB2ZXJzaW9uPSIxLjEiIHg9IjBweCIgeT0iMHB4IiB2aWV3Qm94PSIwIDAgNDA1LjQ1NiA0MDUuNDU2IiBzdHlsZT0iZW5hYmxlLWJhY2tncm91bmQ6bmV3IDAgMCA0MDUuNDU2IDQwNS40NTY7IiB4bWw6c3BhY2U9InByZXNlcnZlIiB3aWR0aD0iMTZweCIgaGVpZ2h0PSIxNnB4Ij4NCjxnPg0KCTxwYXRoIGQ9Ik03NC4xMzQsNjQuMTQ3Yy00Ljk4NSwwLjA3OC05LjkxMSwyLjE2My0xMy40MzgsNS42ODhsLTU1LDU1QzIuMDk2LDEyOC40MzIsMCwxMzMuNDkyLDAsMTM4LjU4MyAgIHMyLjA5NiwxMC4xNTEsNS42OTcsMTMuNzVsMTgzLjI4MSwxODMuMjgxYzMuNTk5LDMuNjAxLDguNjU5LDUuNjk3LDEzLjc1LDUuNjk3czEwLjE1MS0yLjA5NiwxMy43NS01LjY5N2wxODMuMjgxLTE4My4yODEgICBjMy42MDEtMy41OTksNS42OTctOC42NTksNS42OTctMTMuNzVzLTIuMDk2LTEwLjE1MS01LjY5Ny0xMy43NWwtNTUtNTVjLTMuNTk4LTMuNTkxLTguNjUxLTUuNjgxLTEzLjczNC01LjY4MSAgIGMtNS4wODMsMC0xMC4xMzYsMi4wOS0xMy43MzQsNS42ODFMMjAyLjcyOCwxODQuMzk3TDg4LjE2Niw2OS44MzNDODQuNDk5LDY2LjE2OSw3OS4zMTgsNjQuMDcsNzQuMTM0LDY0LjE0N0w3NC4xMzQsNjQuMTQ3eiIgZmlsbD0iIzk2OTY5NiIvPg0KPC9nPg0KPC9zdmc+DQo=);

  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.primary};
  }
`

const Select = ({ name, onChange, options }) => (
  <SelectStyle onChange={onChange} name={name}>
    {options.map(obj => (
      <Option key={obj.key} value={obj.value}>
        {obj.value}
      </Option>
    ))}
  </SelectStyle>
)

export default Select
