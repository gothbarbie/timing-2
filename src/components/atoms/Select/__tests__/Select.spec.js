import React from 'react'
import { shallow } from 'enzyme'

import Select from '../Select'

describe('components/atoms/Select', () => {
  it('renders', () => {
    const options = [
      {
        key: 'test',
        value: 'test',
      },
    ]
    const component = shallow(<Select name="test" options={options} />)

    expect(component).toMatchSnapshot()
  })
})
