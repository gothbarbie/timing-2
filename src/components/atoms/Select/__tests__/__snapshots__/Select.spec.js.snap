// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Select renders 1`] = `
<SelectStyle
  name="test"
>
  <Option
    key="test"
    value="test"
  >
    test
  </Option>
</SelectStyle>
`;
